raiden-cli/src/utils/logging.ts

```diff
@@ -0,0 +1,22 @@
+import log, { Logger, methodFactory, LoggingMethod } from 'loglevel';
+
+export function setupLoglevel(): Logger {
+  log.methodFactory = (
+    methodName: string,
+    level: 0 | 1 | 2 | 3 | 4 | 5,
+    loggerName: string,
+  ): LoggingMethod => {
+    const rawMethod = methodFactory(methodName, level, loggerName);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return (...message: any[]): void => {
+      const prefix = `${new Date(Date.now()).toISOString()} [${methodName}]`;
+      rawMethod(prefix, ...message);
+    };
+  };
+  if (process.env.NODE_ENV === 'production') {
+    log.setLevel('INFO');
+  } else {
+    log.setLevel('DEBUG');
+  }
+  return log;
```

Comment:
Careful: you're actually mutating and returning the global instance/root logger here. It's ok to have it in a function you're sure is called only once (e.g. on beginning of main, as you do), but please, don't return nor use it directly in the `cli`. Instead, return `void` here, and make `log` there be a `log: logging.Logger = logging.getLogger(<some string unique per cli based on address>);`. Also, if possible, for consistency, rename `log` here to `logging`, so we remember this is the root logger and not some child logger.

Reply:
:heavy_check_mark: